### ISSUE TITLE
Number of replicates in `Settings` object

### ISSUE THREAD

**Issue by ijpulidos** [OpenFreeEnergy/gufe#122]
While working on the [NEQ cycling protocol in perses](https://github.com/choderalab/perses/pull/1066) it was noted that we might want to run many cycles/replicates of the simulation unit of the protocol. Say we want to run 100 cycles of a transformation and get a free energy estimate from it.

So far this is stored as a specific setting of the protocol, but this made me wonder if we may want to make this a setting at the gufe `Settings` object level instead of the settings specific to the protocol, since other protocols might benefit from having it.

I guess a benefit from this is that we would then have a standard way of accessing this information if needed and expect protocols implementations to use this setting. It avoids the user having to know what specific setting deals with this and reuse it in different protocols, if desired.

Could this be a setting that we want to have in `gufe` and have protocols to optionally use it?

**Comment by IAlibay** [comment]
I'm not against this, but there is a small language issue that needs to be overcome, one that @richardjgowers likes to annoy me about...

What do we mean by replicates in the wider context outside of an NEQ simulation? In a HREX sense, it's not unlikely that a user may consider this to be the number of replicas along lambda space or the number of replicate simulations sharing a bias in a SAMS simulation, etc...

There is the concept of "repeats" which should definitely be a setting (if it isn't already), and that will likely be universal in meaning, but replicate does end up getting very protocol-specific "meaning".

**Comment by richardjgowers** [comment]
We currently call this `n_repeats`, and it generates that many completely independent runs of the same process (but with different random seeds):  https://github.com/OpenFreeEnergy/openfe/blob/main/openfe/protocols/openmm_rbfe/equil_rbfe_methods.py#L662

For the non-eq approach, would this also create a duplicate equilibrium sampling run?

**Comment by mikemhenry** [comment]
I think once [NEQ cycling protocol in perses](https://github.com/choderalab/perses/pull/1066) is finished, we can iterate and see what settings we want to pull out. The two steps to this are:

* Functionally, will this be needed by other protocols?
* What do we call it so people know what it means? 

We can create some hierarchies/different protocol settings base classes based on the general protocol class (like a RBFE protocol vs a ABFE protocol might share a FEProtocolSettings base class, but then each of some specific things to that get further customized by the protocol.)

**Comment by ijpulidos** [comment]
> I'm not against this, but there is a small language issue that needs to be overcome, one that @richardjgowers likes to annoy me about...

I agree with you about the language issue and what do we exactly mean by replicates, I even had an issue trying to explain this to others when I was working on using "replicates" for the NEQ cycling protocol.

> There is the concept of "repeats" which should definitely be a setting (if it isn't already), and that will likely be universal in meaning, but replicate does end up getting very protocol-specific "meaning".

I don't think we have something like that in the settings models. What would "repeats" mean exactly in this context? Just re-run the protocol for `n_repeated` amount of times?

**Comment by ijpulidos** [comment]
> I think once [NEQ cycling protocol in perses](https://github.com/choderalab/perses/pull/1066) is finished, we can iterate and see what settings we want to pull out. The two steps to this are:

Yeah, this might seem like a premature optimization of the settings, I agree. We can reiterate on this issue once we have fully functional implementations of the protocols.
